src/detect-app-layer-event.c

```diff
@@ -286,12 +286,22 @@ static int DetectAppLayerEventSetup(DetectEngineCtx *de_ctx, Signature *s, const
     sm->ctx = (SigMatchCtx *)data;
 
     if (event_type == APP_LAYER_EVENT_TYPE_PACKET) {
-        SigMatchAppendSMToList(s, sm, DETECT_SM_LIST_MATCH);
+        if (SigMatchAppendSMToList(s, sm, DETECT_SM_LIST_MATCH) < 0) {
+            sm->ctx = NULL;
```

Comment:
this is a very verbose pattern added a 100 times or so. Would like to see a more generalized solution.

Reply:
Argh I was already pretty happy to have found a solution :-p

`SigMatchAppendSMToList` can now fail.
And its failure should induce its 100-ish callers to report the failure...

Or could it be that `SigMatchAppendSMToList` still returns `void` but sets a boolean like `sm->errored = true` which can be checked afterward ? (Then I guess you would not know precisely when the overflow happened)

Reply:
Nice, on it :-)